### PR TITLE
Instructor: StudentListPage: Add down button for indication that panel is expandable #7358

### LINKED ISSUE #7358: Instructor: StudentListPage: Add down button for indication that panel is expandable
I think instead of giving a note, there should be a down arrow button for the user to know that panel is expandable. _[screenshot]_
screenshot from 2017-05-24 03-44-37

### PR DESCRIPTION
**Pull request by amarlearning**
Fixes #7358 

Add bootstrap down chevron as an indication that the panel is expandable. _[screenshot]_

_**[Before]**_
screenshot from 2017-05-24 03-44-37

_**[After]**_
screenshot from 2017-05-25 01-19-46

### CODE REVIEW

**Comment by damithc** [comment]
@amarlearning Can they be less conspicuous, like this (from the instructor home page)?
image

**Comment by amarlearning** [comment]
@damithc doing the same

**Comment by amarlearning** [comment]
No wonder why there is no Chevron. I am facing some problem and need your input -

- The first issue is `toggleChevron` function get called twice. As the user clicks the panel couple of things that happen are [written in series]
    - The `toggleChevron` functions get called which toggles the icon.
    - The [`seeMoreRequest`](https://github.com/TEAMMATES/teammates/blob/3ea7d8e1e5dfcc3417a861d2c790e153a9b1684a/src/main/webapp/js/instructorStudentList.es6#L249) function gets called in which at first the `else if` condition get executed and inside this `else if` condition at the [end we again click the panel via javascript](https://github.com/TEAMMATES/teammates/blob/3ea7d8e1e5dfcc3417a861d2c790e153a9b1684a/src/main/webapp/js/instructorStudentList.es6#L317) because of which the `toggleChevron` functions get called again which changes the icon again.

- The `toggleChevron` function not only changes the Chevron icon but also changes the other icons present inside the parent element. 

It's just an update.

**Comment by amarlearning** [comment]
@damithc ready for review

**Comment by damithc** [comment]
@amarlearning can you post an updated screenshot?

File: src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag [hunk 1]
```diff
@@ -5,7 +5,7 @@
 <%@ attribute name="index" required="true" %>
 <c:set var="PANEL_TYPE" value="${course.courseArchived ? 'panel-default' : 'panel-info'}" />
 <div class="panel ${PANEL_TYPE}">
-    <div class="panel-heading ajax_submit">
+    <div class="panel-heading ajax_submit" onclick="toggleChevron(this)">
```
Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag]: Is there a solution that can avoid using `onclick`?

Comment (amarlearning) [inline src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag]: Other than `onclick` either we can use the `addEventListener` method or we can also use the `on` event with respect to `click`.

**Comment by amarlearning** [comment]
_[Updated Screenshot]_
screenshot from 2017-06-02 18-42-42

File: src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag (hunk 1 above)
Comment (amarlearning) [inline src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag]: @wkurniawan07 should i change my solution?

Comment (wkurniawan07) [inline src/main/webapp/WEB-INF/tags/instructor/studentList/studentsTable.tag]: @amarlearning you should implement a solution that does not involve any JS inside the JSP/tag/HTML files

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/js/common.es6 [hunk 2]
```diff
@@ -988,7 +988,7 @@ function setChevronToDown(chevronContainer) {
 function toggleChevron(clickedElement) {
     const $clickedElement = $(clickedElement);
     const isChevronDown = $clickedElement.find('.glyphicon-chevron-down').length > 0;
-    const $chevronContainer = $clickedElement.find('.glyphicon');
+    const $chevronContainer = $clickedElement.find('.glyphicon-chevron-up, .glyphicon-chevron-down');
```
Comment (wkurniawan07) [inline src/main/webapp/js/common.es6]: What is the reason for this change?

File: src/main/webapp/js/instructorStudentList.es6 [hunk 3]
```diff
@@ -249,6 +249,7 @@ function removeDataToBeTransported() {
 const seeMoreRequest = function (e) {
     const panelHeading = $(this);
     const panelCollapse = $(this).parent().children('.panel-collapse');
+    const toggleCheveron = $(this).parent().find('.glyphicon-chevron-down, .glyphicon-chevron-up');
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorStudentList.es6]: `Cheveron` -> `Chevron`

File: src/main/webapp/js/instructorStudentList.es6 [hunk 4]
```diff
@@ -323,6 +324,12 @@ const seeMoreRequest = function (e) {
         // Do not make ajax call if students shown already above limit
         showStudentLimitError(courseCheck, displayIcon);
     }
+
+    if ($(panelCollapse).attr('class').indexOf('checked') === -1) {
+        $(toggleCheveron).addClass('glyphicon-chevron-down').removeClass(' glyphicon-chevron-up');
+    } else {
+        $(toggleCheveron).addClass('glyphicon-chevron-up').removeClass(' glyphicon-chevron-down');
```
Comment (wkurniawan07) [inline src/main/webapp/js/instructorStudentList.es6]: Remove the space in `glyphicon-chevron-down`. Similarly for `glyphicon-chevron-up` up there.

Also, see if there are chevron-related JS functions defined somewhere that you can reuse.

File: src/main/webapp/js/common.es6 (hunk 2 above)
Comment (amarlearning) [inline src/main/webapp/js/common.es6]: Let's assume this is the sample code:
``` HTML
<div class="parent" onclick="toggleChevron(this)">
    <span class="glyphicon glyphicon-chevron-down">
    <span class="glyphicon glyphicon-search">
</div>
```
As you can see we have two element with class name `glyphicon` inside the parent `div`. Everytime the user clicks on the parent div, the `toggleChevron` function will be called and it will look for all elements with class name `glyphicon` inside the parent div and will toggle classes of thoes elements with class name `glyphicon`.

**_Solution:_**
Instead of looking for elements with class name `glyphicon`, we should look for elements with the class name either `glyphicon-chevron-up` or `glyphicon-chevron-down`.

File: src/main/webapp/js/instructorStudentList.es6 (hunk 4 above)
Comment (amarlearning) [inline src/main/webapp/js/instructorStudentList.es6]: @wkurniawan07 I tried to use the `toggleChevron` function defined inside [common.es6](https://github.com/TEAMMATES/teammates/blob/master/src/main/webapp/js/common.es6#L988), but can't use that because whenever the user click on the `panel-heading` of div, the `seeMoreRequest` function is called which again triggers the `panel-heading` of div due to which the `toggleChevron` function is called twice in just one click of user on `panel-heading`. 

That simply means the `toggleChevron` is called twice even when the user clicks on the panel just one time.

**Comment by amarlearning** [comment]
ready for review

File: src/main/webapp/js/common.es6 (hunk 2 above)
Comment (wkurniawan07) [inline src/main/webapp/js/common.es6]: It is indeed a concern, however I do not see the direct reason why this should be included in this PR. However, since it is a small change, I'll let it pass.
